### PR TITLE
DOCSP-38469 Add OpenAPI Docs

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

This PR demonstrates what the OpenAPI spec looks like in the Relational Migrator documentation. 

> [!NOTE]  
> There is currently a bug in snooty, where the breadcrumb does not render. There will be a Back to Relational Migrator Docs link in the left-hand side nav when the site is deployed to production.

## STAGING

https://preview-mongodbianfmongodb.gatsbyjs.io/docs-relational-migrator/DOCSP-38469/api-docs/

https://preview-mongodbianfmongodb.gatsbyjs.io/docs-relational-migrator/DOCSP-38469/api-docs/rest-api-spec/

## JIRA

https://jira.mongodb.org/browse/DOCSP-38469

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=662a7780b67dbf804a3263a6

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

**Comment by ianf-mongodb** [comment]
Hi @tomhollander, this PR adds OpenAPI documentation to the site. Do you have any suggestions for the name of the page or anything you would like to tweak? Please note staging currently has a bug, which does not render a back button from the actual API docs page. This will not be the case when the site is deployed to production.

**Comment by tomhollander** [comment]
Thanks. The rendering of the content looks good overall. The ToC entry should say "REST API" not "Rest API".
The content itself needs some work, e.g. we need an overview explaining the purpose of the API and how to call it (i.e. it runs against the local RM instance, is governed by the host adapter binding, does not support authentication, etc) as well as more detailed descriptions of each endpoint. I presume that actually improving the content is out of scope of this PR though?

**Review by ianf-mongodb — COMMENTED** [review]
@tomhollander The actual definitions on the spec page will need to be handeled in the application repo, but we do have some flexibility in the way we present and stage the content. I borrowed some copy from https://www.mongodb.com/docs/atlas/app-services/data-api/ and adapted for Relational Migrator. Consider this a rough draft, but I am attempting to introduce and touch on the previous points you mentioned. Please have a look and let me know what you think of this additional stub page.

File: source/api-docs.txt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- You generally prefer scripting your workflows instead of using a GUI.
+
+Constraints
+-----------
+
+The following constraints apply to Data API requests and responses: 
+
+- Requests will time out after 300 seconds of runtime.
+
+- API response payloads are limited to 350 MB in size.
```
Comment (ianf-mongodb) [inline source/api-docs.txt:67]: Borrowed these from the atlas docs, please LMK if there are additional constraints.

Comment (tomhollander) [reply source/api-docs.txt:67]: I would skip this the mention of payload size; while I'm sure it's technically true there is no reason the payloads would be anywhere near this size for this API.

**Review by tomhollander — APPROVED** [review]

File: source/api-docs.txt [hunk 2]
```diff
@@ -0,0 +1,69 @@
+.. _rm-rest-api:
```
Comment (ianf-mongodb) [inline source/api-docs.txt:1]: **Note to Internal Review**

This is the only real page that needs to be reviewed, the spec file itself is maintained in another repo and should not be considered part of this review.

File: source/api-docs.txt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+The Relational Migrator REST API provides you with programmatic access 
+to Relation Migrator with HTTPS endpoints. The REST API is a middleware 
+service that sits between your application and your source and destination
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:23]: Avoid "sit" (not global)
Also, use something other than "middleware" if you can.

File: source/api-docs.txt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+   :class: singlecol
+
+The Relational Migrator REST API provides you with programmatic access 
+to Relation Migrator with HTTPS endpoints. The REST API is a middleware 
+service that sits between your application and your source and destination 
+databases that you can send requests to facilitate and manage your sync 
+jobs.
+
+You don't need to install any additional extensions or packages to use 
+the Relation Migrator REST API. The functionality is built into the core
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:28]: Can you avoid "core"?
And, "built into"? Sounds a bit too "marketing informal". I'd go with something like "The Relational Migrator REST API is part of the Relational Migrator product..." or something like that.

File: source/api-docs.txt [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+The Relational Migrator REST API provides you with programmatic access 
+to Relation Migrator with HTTPS endpoints. The REST API is a middleware 
+service that sits between your application and your source and destination 
+databases that you can send requests to facilitate and manage your sync 
+jobs.
+
+You don't need to install any additional extensions or packages to use 
+the Relation Migrator REST API. The functionality is built into the core 
+Relational Migrator product and runs on the same port that you access
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:29]: I'd split the statement into 2. Put the port items in the 2nd.

File: source/api-docs.txt [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+The Relational Migrator REST API provides you with programmatic access 
+to Relation Migrator with HTTPS endpoints. The REST API is a middleware 
+service that sits between your application and your source and destination 
+databases that you can send requests to facilitate and manage your sync
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:24]: I'd remove "facilitate". Too informal.

File: source/api-docs.txt (hunk 4 above)
Comment (jason-price-mongodb) [inline source/api-docs.txt:28]: Typo: "Relation ..."

File: source/api-docs.txt [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+the application on (either ``8278``, ``8080``, or ``443``).
+
+How the REST API Works
+----------------------
+
+The Relational Migrator REST API:
+
+- Runs on the machine or server that the Relation Migrator application
+  is installed on.
+- The Rest API is governed by the host adapter binding.
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:39]: Avoid "governed".

File: source/api-docs.txt [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+the Relation Migrator REST API. The functionality is built into the core 
+Relational Migrator product and runs on the same port that you access 
+the application on (either ``8278``, ``8080``, or ``443``).
+
+How the REST API Works
+----------------------
+
+The Relational Migrator REST API:
+
+- Runs on the machine or server that the Relation Migrator application
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:37]: Use "computer", not "machine". Also, is it "client computer"?

File: source/api-docs.txt [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+- The Rest API is governed by the host adapter binding.
+- The Relational Migrator REST API **does not** currently support 
+  authentication.
+
+When to Use REST API
+--------------------
+
+We recommend using the Relational Migrator REST API when:
+
+- You want to use a task scheduler such as
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:48]: Split long complex statement.

File: source/api-docs.txt [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+We recommend using the Relational Migrator REST API when:
+
+- You want to use a task scheduler such as 
+  `CRON <https://en.wikipedia.org/wiki/Cron>`__ and define dynamic 
+  :ref:`table filters <rm-table-filters>` within a project such as 
+  ``WHERE DATEDIFF(CURDATE(), last_modified) <= 1`` to batch migrate 
+  data on a cadence from your relational database to MongoDB. 
+
+- You want to programmatically create Relational Migrator sync jobs 
+  quickly and dynamically. For example, using connection strings from
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:55]: Bit too informal: "quickly and dynamically". I'd rewrite this.

File: source/api-docs.txt [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+   :description: Learn about the MongoDB Relational Migrator REST API and how to use it.
+   :keywords: data migration, rest api, open api
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+The Relational Migrator REST API provides you with programmatic access
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:21]: Consider rewrite into format "To (state goal), (state action)" if you can. Cheers.

File: source/api-docs.txt [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+- Runs on the machine or server that the Relation Migrator application
+  is installed on.
+- The Rest API is governed by the host adapter binding.
+- The Relational Migrator REST API **does not** currently support 
+  authentication.
+
+When to Use REST API
+--------------------
+
+We recommend using the Relational Migrator REST API when:
```
Comment (jason-price-mongodb) [inline source/api-docs.txt:46]: Consider rewrite "Use the ... when:"

**Review by jason-price-mongodb — APPROVED** [review]
@ianf-mongodb overall, looks good, but some changes needed. Lgtm'd it, but let me know if you want me to take another look after you're done. But, if the changes are implemented, it should be good. Cheers.

**Comment by davidsoulmdb** [comment]
Minor suggestion: The bolding on 'The Relational Migrator REST API does not currently support authentication' makes it a strong focus, perhaps does not need to be emphasised heavily?

**Comment by tomhollander** [comment]
> Minor suggestion: The bolding on 'The Relational Migrator REST API does not currently support authentication' makes it a strong focus, perhaps does not need to be emphasised heavily?

I think this is pretty important thing for people to know.

**Comment by ianf-mongodb** [comment]
> > Minor suggestion: The bolding on 'The Relational Migrator REST API does not currently support authentication' makes it a strong focus, perhaps does not need to be emphasised heavily?
> 
> I think this is pretty important thing for people to know.

Ok I will leave this as is for now.

**Pull request merged**